### DISCUSSION TITLE
natural query language

### DISCUSSION THREAD

**Discussion by dridk** [meilisearch/product#600] [Feedback & Feature Proposal]
Hi, 

It would be greate to have a natural query language like tantivy search engine to perform query and filter in the same line of text. 

for instance : 

"title:boby AND body:test OR ( age>10 AND age age < 28)"

I know it is actually possible from a json query using the filter key. but How can I do it from a HTML search bar ? How can translate the query above to the json query ?

**Comment by oluademola** [comment]
Hello @dridk 

Have you tried building a filter expression with the examples provided in the dedication [documentation](https://docs.meilisearch.com/learn/advanced/filtering_and_faceted_search.html#filter-expressions)? Is there anything that did not work? You can create a filter expression like this: `"filter": "title = boby AND body = test OR ( age > 10 AND age < 28)"` Could you please try and let us know? Thanks

**Reply by dridk** [reply]
Filters works. 
I am asking something like google search bar where you can type an expression like: 

"cheval type:pdf" 

Or like pubmed search Bar 

"title:cardio AND abstract:hearth"
